modules/configuration/php/NDB_Form_configuration.class.inc

```diff
@@ -135,6 +137,29 @@ class NDB_Form_Configuration extends NDB_Form
                'optimal' => 'Optimal Visit Window for Visit Label',
               );
     }
+
+    /**
+    * Loads the project management module
+    *
+    * @return none
+    */
+    function project()
+    {
+        $factory     = NDB_Factory::singleton();
+        $config      = $factory->config();
+        $configs     = NDB_Config::singleton();
+        $useProjects = $config->getSetting("useProjects");
+        if ($useProjects === "true") {
+            $projectList = Utility::getProjectList();
+            $projects    = array();
+            foreach ($projectList as $ProjectID => $Name) {
+                $projects[$ProjectID] = $config->getProjectSettings($ProjectID);
+            }
+            $this->tpl_data['projects'] = $projects;
+        } else {
+            throw new Exception("Projects are not used in this study.");
```

Comment:
ConfigurationException makes sense. Basically, the user tried to access a
page which he does not use.
On 28 Aug 2015 10:21 am, "Dave MacFarlane" <notifications@github.com> wrote:

> In modules/configuration/php/NDB_Form_configuration.class.inc
> <https://github.com/aces/Loris/pull/1220#discussion_r38204246>:
>
> > +    */
> > +    function project()
> > +    {
> > +        $factory     = NDB_Factory::singleton();
> > +        $config      = $factory->config();
> > +        $configs     = NDB_Config::singleton();
> > +        $useProjects = $config->getSetting("useProjects");
> > +        if ($useProjects === "true") {
> > +            $projectList = Utility::getProjectList();
> > +            $projects    = array();
> > +            foreach ($projectList as $ProjectID => $Name) {
> > +                $projects[$ProjectID] = $config->getProjectSettings($ProjectID);
> > +            }
> > +            $this->tpl_data['projects'] = $projects;
> > +        } else {
> > +            throw new Exception("Projects are not used in this study.");
>
> Should be a LorisException (or maybe a ConfigurationException?)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aces/Loris/pull/1220/files#r38204246>.
>